### DISCUSSION TITLE
Name of the user who created/uploaded the folder/file

### DISCUSSION THREAD

**Discussion by GeorgeGeorgeGe** [rejetto/hfs#271] [Ideas]
Hi,

Would you please let me if it is possible to display the Name of the user who created/uploaded the folder/file in a Folder List?

Thanks in advance.

**Comment by rejetto** [comment]
This information is not stored, at the moment.
You may be able to retrieve it from the log, but logs are supposed to be deleted at will, so I cannot count on it (and searching a log is not efficient, anyway).
So, first problem for the feature you ask is WHERE I should store the information.
This is a non-trivial problem, because I have some options but there are downsides to each.
I will have to study.

**Comment by GeorgeGeorgeGe** [comment]
I think you can get the information from the file system of the physical storage. the same way you get the modified date.
what do you think?
Thanks.

**Comment by rejetto** [comment]
"user who uploaded" is not written in the file system

**Reply by GeorgeGeorgeGe** [reply]
he is the "Author" i think.

**Reply by rejetto** [reply]
I'm not sure I get what you mean.
Where are you seeing the "author" in the file system?

**Reply by GeorgeGeorgeGe** [reply]
in windows, in windows explorer, in the folder that contains that file/folder.

**Reply by rejetto** [reply]
I tried to look at a random file in my Windows machine, and I don't see any "author" field

**Reply by GeorgeGeorgeGe** [reply]
the additional columns is hidden by default, right click on any empty space on the top bar of the list and choose the column you want to add.

**Reply by rejetto** [reply]
<img width="193" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/dc964ad7-7374-460a-a8f3-1fed4314ff88">

so you are talking about these...
"assistant's name" ?
I don't think these are real. 
My guess: it's info possibly stored inside the file itself and depends on the file format.
Example: you see a PDF in the list, Acrobat Reader installs a system extension that makes Windows see extra attributes that are actually stored inside the PDF, not a real file attribute.

I'm already looking [a similar thing](https://github.com/mafintosh/fs-extended-attributes), but it must be cross-platform.
I already studied for like 1 hour on the problem. It won't be a quick thing.

**Reply by GeorgeGeorgeGe** [reply]
some of my users delete the original file after modify it on their pc and upload the new one, other users unable to know who did the last modifications.
is there any way to get around this issue? please advice.

**Reply by rejetto** [reply]
as I said in my first post, you may be able to find this information in the logs.
I can't see another place.

**Reply by GeorgeGeorgeGe** [reply]
Thanks a lot.

**Comment by TEA-Time1** [comment]
Massimo is correct.  A lot of those in that list cause add-ons to File Explorer to pull that info from metadata in certain types of files (Word and Excel are a couple other examples) and are not stored in the filesystem itself.

-Tim

________________________________________

**Reply by GeorgeGeorgeGe** [reply]
Thanks TEA-Time 1, i thought that displaying at least the "Authors" Field may resolve part of my problem.
Thank you for clarification.

**Reply by TEA-Time1** [reply]
👍

**Comment by rejetto** [comment]
What do you think of something like this?

<img width="265" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/08972c9a-44f7-4eaa-88dd-209daa8de22e">

**Reply by GeorgeGeorgeGe** [reply]
That would be perfect and very useful for me.
Thanks.

**Reply by rejetto** [reply]
available in beta8 #257

**Reply by GeorgeGeorgeGe** [reply]
it works perfect !
But, (in the next version) can we make it displayed in the list as well, beside the file size and the timestamp  ? 
Thanks a lot.

**Reply by rejetto** [reply]
i'm not sure, it can take a lot of space in the list... can you imagine a solution to this?

**Reply by GeorgeGeorgeGe** [reply]
I think it does not need that much space to be displayed in the list , it is only the user name field that will be added there.
if you want you can make the display of each field of the list (except the file name) to be optional by the user.
Thanks.

**Reply by rejetto** [reply]
nope, it's IP + username. Username is not always available, you may have upload permissions for anonymous.
And you also need at least an icon to indicate WHAT you are showing, or worse the "uploader:" text.
So it could be like "uploader: 12.34.56.78 myusername"
it's not short.

**Reply by GeorgeGeorgeGe** [reply]
no need for the "uploader" as a text before the username, what about [IP address]-[username] ?

**Reply by rejetto** [reply]
no need? ANd how will people understand what is the ip-username about?

**Reply by GeorgeGeorgeGe** [reply]
the public IP address, most of the time, it is dynamic , no use of it, i think.
Thanks

**Reply by GeorgeGeorgeGe** [reply]
Currently, you, optionally, display the download counter without explanation of what it does mean.
Thanks

**Reply by rejetto** [reply]
there's an icon with a tooltip (and that's a plugin, quality standards are a bit lower)

**Reply by GeorgeGeorgeGe** [reply]
i agree with you, but repeating the text "uploader" on every line of the list is not the best option.
you can add a (*) symbol with a note somewhere explain what does that filed contains in details.
Regards

**Reply by rejetto** [reply]
we can use an icon, of course, and we and up with "⬆ 12.34.56.78 myusername"
still a bit long.
Maybe I can make it a plugin

**Reply by GeorgeGeorgeGe** [reply]
Great idea!

**Reply by GeorgeGeorgeGe** [reply]
i think no need for the ip address, you can get nothing from it.

**Reply by rejetto** [reply]
you are considering your own case, where you probably have restricted upload to (some) accounts.
I have to consider EVERY case, and it is possible in HFS to make upload available to anonymous users.

**Reply by rejetto** [reply]
i may have an option in the plugin about what you want to show

**Reply by rejetto** [reply]
like this
<img width="285" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/63e390bb-d623-41b8-93a5-7dd40078edd8">

**Reply by GeorgeGeorgeGe** [reply]
Perfect.

**Reply by rejetto** [reply]
check #257

**Reply by GeorgeGeorgeGe** [reply]
It works perfect.
Thank you very very much.

**Reply by GeorgeGeorgeGe** [reply]
very small Issue:
if you run the plugin list-uploader then a refresh is needed after you upload a file and then rename it to display the list properly.
Please check.
Thanks.

**Reply by GeorgeGeorgeGe** [reply]
can we make the list-uploader works on folders also, now i think it works on file only ?
Thanks

**Comment by rejetto** [comment]
> if you run the plugin list-uploader then a refresh is needed after you upload a file and then rename it to display the list properly.

I see the problem, but it has nothing to do with upload. Any renamed file is affected. Fixed in next release

**Comment by rejetto** [comment]
> can we make the list-uploader works on folders also, now i think it works on file only ?

the problem is not in the plugin. No "uploader" information is stored on folders. Fixed in next release

**Comment by rejetto** [comment]
new beta pubished #257

**Reply by GeorgeGeorgeGe** [reply]
it works PERFECT!
Thanks a lot.

**Comment by GeorgeGeorgeGe** [comment]
Hi,
Please note that the first visit to a page does not display the users who uploaded the files, after refreshing the page from the browser refresh button, they are shown.
the uploader plugin is not working on Folders.
Thanks

**Reply by rejetto** [reply]
i was unable to reproduce your problem with neither chrome or firefox.
if you think you've found a bug, please open an issue and provide all the information
https://github.com/rejetto/hfs/issues/new?assignees=&labels=bug&projects=&template=bug_report.md
